.github/workflows/main.yml

```diff
@@ -0,0 +1,302 @@
+name: Build and Test
```

Comment:
From our discussion yesterday it sounded like the problem for Windows was that this configuration turns off compiler-guided inlining by default, otherwise the optimisation level is identical between `RelWithDebInfo` and `Release` on Windows (`/O2`). I have an update to change this which also documents the discussed rationale from the chat.

Reply:
I have updated this to use Release for now in CI.